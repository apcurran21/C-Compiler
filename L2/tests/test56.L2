(@main
  (@main
0
    :call_label0
    %test_value <- 1234
    %newVar1 <- 9
    %newVar1 >>= 1
    %newVar0 <- %newVar1
    %newVar2 <- 9
    %newVar2 >>= 1
    %newVar0 <- %newVar0
    %newVar0 *= %newVar2
    %newVar3 <- 9
    %newVar3 >>= 1
    %newVar0 <- %newVar0
    %newVar0 *= %newVar3
    %newVar0 <- %newVar0
    %newVar0 <<= 1
    %newVar0 <- %newVar0
    %newVar0 += 1
    %newVar0 <- %newVar0
    %newVar0 += 8
    rdi <- %newVar0
    rsi <- 1
    call allocate 2
    %matrix <- rax
    %newVar4 <- %matrix
    %newVar4 += 8
    mem %newVar4 0  <- 7
    %newVar5 <- %matrix
    %newVar5 += 16
    mem %newVar5 0  <- 9
    %newVar6 <- %matrix
    %newVar6 += 24
    mem %newVar6 0  <- 9
    %newVar7 <- %matrix
    %newVar7 += 32
    mem %newVar7 0  <- 9
    mem rsp -8 <- :call_label1
    rdi <- %matrix
    rsi <- %test_value
    call @find 2
    :call_label1
    %result <- rax
    rdi <- %result
    call print 1
    %newVar9 <- 0
    %newVar10 <- %matrix
    %newVar10 += 24
    %newVar11 <- mem %newVar10 0 
    %newVar12 <- %newVar11
    %newVar12 >>= 1
    %newVar13 <- %matrix
    %newVar13 += 32
    %newVar14 <- mem %newVar13 0 
    %newVar15 <- %newVar14
    %newVar15 >>= 1
    %newVar16 <- 1
    %newVar16 <- %newVar16
    %newVar16 *= %newVar15
    %newVar17 <- 1
    %newVar17 *= %newVar16
    %newVar9 <- %newVar9
    %newVar9 += %newVar17
    %newVar16 <- %newVar16
    %newVar16 *= %newVar12
    %newVar18 <- 2
    %newVar18 *= %newVar16
    %newVar9 <- %newVar9
    %newVar9 += %newVar18
    %newVar9 <- %newVar9
    %newVar9 += 0
    %newVar8 <- %newVar9
    %newVar8 *= 8
    %newVar8 <- %newVar8
    %newVar8 += 40
    %newVar19 <- %matrix
    %newVar19 += %newVar8
    mem %newVar19 0  <- %test_value
    mem rsp -8 <- :call_label2
    rdi <- %matrix
    rsi <- %test_value
    call @find 2
    :call_label2
    %result <- rax
    rdi <- %result
    call print 1
    return
  )
  (@find
2
    %A <- rdi
    %v <- rsi
    :call_label3
    %newVar0 <- 0
    %newVar0 *= 8
    %newVar1 <- %newVar0
    %newVar1 += 16
    %newVar2 <- %A
    %newVar2 += %newVar1
    %length1 <- mem %newVar2 0 
    rdi <- %length1
    call print 1
    %length1 <- %length1
    %length1 >>= 1
    %newVar3 <- 1
    %newVar3 *= 8
    %newVar4 <- %newVar3
    %newVar4 += 16
    %newVar5 <- %A
    %newVar5 += %newVar4
    %length2 <- mem %newVar5 0 
    rdi <- %length2
    call print 1
    %length2 <- %length2
    %length2 >>= 1
    %newVar6 <- 2
    %newVar6 *= 8
    %newVar7 <- %newVar6
    %newVar7 += 16
    %newVar8 <- %A
    %newVar8 += %newVar7
    %length3 <- mem %newVar8 0 
    rdi <- %length3
    call print 1
    %length3 <- %length3
    %length3 >>= 1
    goto :call_label4
    :call_label4
    %found <- 1
    %i <- 0
    %j <- 0
    %k <- 0
    goto :call_label13
    :call_label5
    %i <- %i
    %i += 1
    goto :call_label8
    :call_label6
    %j <- %j
    %j += 1
    goto :call_label9
    :call_label7
    %k <- %k
    %k += 1
    goto :call_label11
    :call_label8
    %j <- 0
    goto :call_label13
    :call_label9
    %k <- 0
    goto :call_label12
    :call_label10
    %newVar10 <- 0
    %newVar11 <- %A
    %newVar11 += 24
    %newVar12 <- mem %newVar11 0 
    %newVar13 <- %newVar12
    %newVar13 >>= 1
    %newVar14 <- %A
    %newVar14 += 32
    %newVar15 <- mem %newVar14 0 
    %newVar16 <- %newVar15
    %newVar16 >>= 1
    %newVar17 <- 1
    %newVar17 <- %newVar17
    %newVar17 *= %newVar16
    %newVar18 <- %j
    %newVar18 *= %newVar17
    %newVar10 <- %newVar10
    %newVar10 += %newVar18
    %newVar17 <- %newVar17
    %newVar17 *= %newVar13
    %newVar19 <- %i
    %newVar19 *= %newVar17
    %newVar10 <- %newVar10
    %newVar10 += %newVar19
    %newVar10 <- %newVar10
    %newVar10 += %k
    %newVar9 <- %newVar10
    %newVar9 *= 8
    %newVar9 <- %newVar9
    %newVar9 += 40
    %newVar20 <- %A
    %newVar20 += %newVar9
    %check_value <- mem %newVar20 0 
    %equal <- %check_value = %v
    
    cjump %equal = 1 :call_label14 
goto :call_label7

    :call_label11
    %check_index <- %k < %length3
    
    cjump %check_index = 1 :call_label10 
goto :call_label6

    :call_label12
    %check_index <- %j < %length2
    
    cjump %check_index = 1 :call_label11 
goto :call_label5

    :call_label13
    %check_index <- %i < %length1
    
    cjump %check_index = 1 :call_label12 
goto :call_label15

    :call_label14
    %found <- 3
    goto :call_label15
    :call_label15
    rax <- %found
    return
  )
)
