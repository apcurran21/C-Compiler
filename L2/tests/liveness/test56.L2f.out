(
(in
(r12 r13 r14 r15 rbp rbx rdi rsi)
(r12 r13 r14 r15 rax rbp rbx rsi)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(r12 r13 r14 r15 rax rbp rbx rsi)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
